src/app/content/components/Sidebar.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import styled, { css } from 'styled-components';
+import { AppState, Dispatch } from '../../types';
+import * as actions from '../actions';
+import { isArchiveTree } from '../guards';
+import * as selectors from '../selectors';
+import { ArchiveTree, Book, Page } from '../types';
+import { stripIdVersion } from '../utils';
+import ContentLink from './ContentLink';
+
+const SidebarPlaceholder = styled.div<{open: boolean}>`
+  background-color: white;
+  overflow-x: hidden;
+  transition: all 300ms;
+  width: 40px;
+
+  ${(props) => props.open && css`
+    width: 300px;
+  `}
+`;
+
+const SidebarControl = styled(({open, ...props}: React.HTMLProps<HTMLButtonElement> & {open: boolean}) =>
+  <button {...props}>
+    <span></span>
+    <span></span>
+    <span></span>
+  </button>
+)`
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  height: 40px;
+  width: 40px;
+  background: none;
+  padding: 0;
+  border: none;
+  z-index: 1;
+  outline: none;
+
+  span {
+    width: 100%;
+    background: #000;
+    padding-top: 10%;
+    display: block;
+    visibility: visible;
+    opacity: 1;
+    transition: all 300ms;
+  }
+
+  ${(props) => !props.open && css`
+    span {
+      margin-top: 15%;
+    }
+  `}
+
+  ${(props) => props.open && css`
+    span:nth-child(1) {
+      transform: rotateZ(45deg);
+    }
+    span:nth-child(2) {
+      opacity: 0;
+    }
+    span:nth-child(3) {
+      transform: rotateZ(-45deg);
+      margin-top: -20%
+    }
+  `}
+
+`;
+
+const SidebarBody = styled.div<{open: boolean}>`
+  position: fixed;
+  left: 0;
+  top: 0;
+  bottom: 0;
+  width: 300px;
+  overflow-y: auto;
+  font-size: 15px;
+  transition: all 300ms;
+  padding: 10px;
+
+  ol {
+    padding-inline-start: 10px;
+  }
+
+  > * {
+    transition-property: visibility;
+    transition-delay: 0s;
+    visibility: visible;
+  }
+
+  ${(props) => !props.open && css`
+    left: -280px;
+    background-color: #ccc;
+    overflow-y: hidden;
+    > :not(${SidebarControl}) {
+      transition-delay: 300ms;
+      visibility: hidden;
+    }
+  `}
+`;
+
+const NavItem = styled.li<{active?: boolean}>`
+  list-style: none;
+  position: relative;
+
+  ${(props) => props.active && css`
+    :before {
+      font-weight: bold;
+      content: '>';
+      position: absolute;
+      right: 100%;
+    }
+  `}
+`;
+
+interface SidebarProps {
+  open: boolean;
+  book?: Book;
+  page?: Page;
+  openToc: typeof actions['openToc'];
+  closeToc: typeof actions['closeToc'];
+}
+
+export class Sidebar extends Component<SidebarProps> {
+
+  public render() {
+    const {open, book, closeToc, openToc} = this.props;
+    return <SidebarPlaceholder open={open}>
+      <SidebarBody open={open}>
+        <SidebarControl
+          open={open}
+          onClick={() => open ? closeToc() : openToc()}
+          role='button'
+          aria-label={`Click to ${open ? 'close' : 'open'} the Table of Contents`}
```

Comment:
Hmm, maybe it's time to open an i18n PR...

Reply:
yea i would prefer to do that before we dig our hole much bigger